numba_dppy/extended_numba_itanium_mangler.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from numba.core import itanium_mangler, types
+from numba_dppy import target
+
+EXQ2ABI = {
+    target.SPIR_PRIVATE_ADDRSPACE: "AS0",
+    target.SPIR_GLOBAL_ADDRSPACE: "AS1",
+    target.SPIR_CONSTANT_ADDRSPACE: "AS2",
+    target.SPIR_LOCAL_ADDRSPACE: "AS3",
+    target.SPIR_GENERIC_ADDRSPACE: "AS4",
+}
+
+
+def mangle_type_or_value(typ):
+    """
+    Mangle type parameter and arbitrary value.
+    This function extends Numba provided `magle_type_or_value()` to
```

Comment:
`Numba's` instead of `Numba provided`? Also, please make the e.g. more verbose. Such as:

For example, an ``int *`` argument will be mangled to "Pi".

Reply:
Updated.